### PR TITLE
handle case when controlnet is list or tuple

### PR DESCRIPTION
**Pull request by noskill**
In StableDiffusionControlNetPipeline constructor  controlnet  has type Union[ControlNetModel, List[ControlNetModel], Tuple[ControlNetModel], MultiControlNetModel] so such case should throw an exception

### CODE REVIEW

**Comment by patrickvonplaten** [comment]
Hey @noskill,

Could you please add a reproducible code snippet that shows the current error ?

**Comment by noskill** [comment]
@patrickvonplaten minimal example:

```
from diffusers import ControlNetModel, StableDiffusionControlNetPipeline


model_id = "realismEngine/realismEngine_v10.safetensors"
cnet_id = "DionTimmer/controlnet_qrcode-control_v1p_sd15"

cnets = [ControlNetModel.from_pretrained(cnet_id)]
StableDiffusionControlNetPipeline.from_single_file(model_id, controlnet=cnets)
```

File: src/diffusers/loaders.py [hunk 1]
```diff
@@ -2337,7 +2337,10 @@ def from_single_file(cls, pretrained_model_link_or_path, **kwargs):
 
             # Model type will be inferred from the checkpoint.
             if not isinstance(controlnet, (ControlNetModel, MultiControlNetModel)):
-                raise ValueError("ControlNet needs to be passed if loading from ControlNet pipeline.")
+                if isinstance(controlnet, (list, tuple)) and controlnet:
+                    pass
+                else:
+                    raise ValueError("ControlNet needs to be passed if loading from ControlNet pipeline.")
```
Comment (patrickvonplaten) [inline src/diffusers/loaders.py:2343]:
```suggestion
```

File: src/diffusers/loaders.py [hunk 2]
```diff
@@ -2336,8 +2336,7 @@ def from_single_file(cls, pretrained_model_link_or_path, **kwargs):
             from .pipelines.controlnet.multicontrolnet import MultiControlNetModel
 
             # Model type will be inferred from the checkpoint.
-            if not isinstance(controlnet, (ControlNetModel, MultiControlNetModel)):
-                raise ValueError("ControlNet needs to be passed if loading from ControlNet pipeline.")
+            if not (isinstance(controlnet, (ControlNetModel, MultiControlNetModel)) or isinstance(controlnet, (list, tuple)) and isinstance(controlnet[0], ControlNetModel))
```
Comment (patrickvonplaten) [inline src/diffusers/loaders.py:2339]:
```suggestion
            if not (isinstance(controlnet, (ControlNetModel, MultiControlNetModel)) or isinstance(controlnet, (list, tuple)) and isinstance(controlnet[0], ControlNetModel)):
                raise ValueError("ControlNet needs to be passed if loading from ControlNet pipeline.")
```

**Comment by patrickvonplaten** [comment]
I see! Indeed that's a very nice fix - thank you :-)

**Pull request merged**